packages/plugin-ext/src/common/plugin-identifiers.ts

```diff
@@ -56,15 +56,15 @@ export namespace PluginIdentifiers {
      * @returns a string in the format `<publisher>.<name>`.
      */
     export function unversionedFromVersioned(id: VersionedId): UnversionedId {
-        const endOfId = id.indexOf('@');
+        const endOfId = id.lastIndexOf('@');
         return id.slice(0, endOfId) as UnversionedId;
     }
     /**
      * @returns `undefined` if it looks like the string passed in does not have the format returned by {@link PluginIdentifiers.toVersionedId}.
      */
     export function identifiersFromVersionedId(probablyId: string): Components | undefined {
         const endOfPublisher = probablyId.indexOf('.');
-        const endOfName = probablyId.indexOf('@', endOfPublisher);
+        const endOfName = probablyId.lastIndexOf('@');
```

Comment:
I can imagine this fail on versions like `@acme/publisher.plugin`.

Maybe a quick `.spec.ts` file testing these functions could help here.

Reply:
The description of this method says `if it looks like the string passed in does not have the format returned by {@link PluginIdentifiers.toVersionedId}.`. [As far as I know the format it refers to is `${string}.${string}@${string}`](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/common/plugin-identifiers.ts#L29) so it an unversioned id name is passed it should not be supported. With the change coming from this PR if an unversion id is passed the method would not return undefined and this might indeed be a problem.

As far as I understand with scoped packages the organization name can be pretty much any identifier (even only numbers and dots) so discriminating between that and and a version would require to look at what characters are allowed in package names and in versions. [The website `semver.org` offers a regex to test if a version is valid](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string), so it might make sense to use that to test whether a version is valid.

Following this idea:

- `publisher.package-name@3.5.6` => valid
- `publisher.@3.5.6/package-name@3.5.6` => valid
- `publisher.@3.5.6` => valid (name is empty, are we happy with that? ...probably not)
- `publisher.package-name@3.5.6/name` => not valid (version cannot contain `/`)

If you are happy with this idea I'm happy to improve the information extraction according to this logic.